### PR TITLE
Store() function

### LINKED ISSUE #637: Store bitmap result directly
### Description
There is currently no way to compute a bitmap query and store it directly as a new row. 

Example: `Store(<bitmap_query>, frame=foo, rowID=10)` -> `{"results":[{"rowID": 10}]}`
(maybe rowID input is optional)

### Success criteria (What criteria will consider this ticket closeable?)
Any query that returns a bitmap result can be immediately written to a new frame+row, without returning the full bitmap result to the client.

### PR DESCRIPTION
**Pull request by travisturner**
## Overview

This PR implements `Store(<row_query>, <fieldname>=10)`. Currently, it's only supported on `set` fields. It's not clear to me yet how we would support this on `mutex` and `bool` fields. And I don't think it makes sense to store a `row` result into `int` or `time` fields (although I suppose if `Store()` took an optional timestamp, it could apply the row to all time quantums).

Also, the PQL function is `Store()`, but internally everything is `SetRow()` because it seemed like a complement to `ClearRow()`.

Fixes #637 (at least the first phase of it)

TODO:
- [x] Docs
- [x] Decide if we want to support `time` fields.

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: executor.go [hunk 1]
```diff
@@ -1213,6 +1215,98 @@ func (e *executor) executeClearRowShard(_ context.Context, index string, c *pql.
...
+	fieldName, err := c.FieldArg()
+	if err != nil {
+		return false, errors.New("SetRow() argument required: field")
+	}
+	field := e.Holder.Field(index, fieldName)
+	if field == nil {
+		return false, ErrFieldNotFound
+	}
+
+	switch field.Type() {
```
Comment (tgruben) [inline executor.go]: I would prefer an if for a one state check
 ` if field.Type() != FieldTypeSet{
  return false, fmt.Errorf("SetRow() is not supported on %s field types", field.Type())
}`

File: fragment.go [hunk 2]
```diff
@@ -491,6 +491,56 @@ func (f *fragment) unprotectedClearBit(rowID, columnID uint64) (changed bool, er
...
+	// First container of the row in storage.
+	headContainerKey := rowID << shardVsContainerExponent
+
+	// Remove every existing container in the row.
+	for i := uint64(0); i < (1 << shardVsContainerExponent); i++ {
+		f.storage.Containers.Remove(headContainerKey + i)
+	}
+
+	// From the given row, get the rowSegment for this shard.
+	seg := row.segment(f.shard)
```
Comment (tgruben) [inline fragment.go]: i think this check needs to be before the remove

Comment (travisturner) [reply fragment.go]: In a multi-shard scenario, if you don't remove the existing row, you could potentially have updated data on one shard, and old data on another.

File: executor.go (hunk 1 above)
Comment (travisturner) [reply executor.go]: Yeah, this makes sense. I left it this way in the case that I was going to modify this PR to include the `time` field type.

**Comment by jaffee** [comment]
I vote "no" on time fields - keep things simple for now.

**Comment by tgruben** [comment]
I vote only set fields

**Comment by travisturner** [comment]
Since we're only supporting `set` fields as the destination field, I'm taking this PR out of WIP.

**Comment by alanbernstein** [comment]
This supports an arbitrary bitmap call, right? I would suggest adding a second docs example explicitly showing something simple like `Store(Intersect(x, y), z)`, to demonstrate that you can sort of perform computations this way. Maybe a test that does something like that as well.

I agree on only supporting `set` fields for now, but some thoughts on future directions... Storing to `mutex` seems reasonable - doesn't it make sense (conceptually at least) to write some bits to one row of a `mutex` field, then clear any conflicting bits as necessary? 

I can imagine performing arithmetic operations on `int` fields and storing the entire result into another `int` field. This is obviously far-future kind of work though, and its not clear if it would even make sense to use Pilosa for this.

**Comment by travisturner** [comment]
@alanbernstein I added another example to the query docs.

I think that `mutex` support is do-able, but it will likely require that we handle the individual columns, checking for existing values on each.

I like the idea of storing arithmetic operations performed on `int` fields. That made me wonder about doing arithmetic operations on an `int` field in place. For example, add 7 to all columns in `int` field `f`. I think this would require us to do multiple passes and re-writing of rows, and I'm not even sure that's useful, but something we could likely do fairly easily.

**Pull request merged**